bundle/edu.gemini.spModel.pio/src/main/scala/edu/gemini/spModel/pio/codec/ParamCodec.scala

```diff
@@ -0,0 +1,59 @@
+package edu.gemini.spModel.pio.codec
+
+import scalaz._, Scalaz._
+
+import edu.gemini.spModel.pio._
+import edu.gemini.spModel.pio.xml.{ PioXmlUtil, PioXmlFactory }
+
+trait ParamCodec[A] { outer =>
+  def encode(key: String, a: A): Param
+  def decode(p: Param): PioError \/ A
+  def xmap[B](f: A => B, g: B => A): ParamCodec[B] =
+    new ParamCodec[B] {
+      def encode(key: String, a: B): Param = outer.encode(key, g(a))
+      def decode(p: Param): PioError \/ B = outer.decode(p).map(f)
+    }
+}
+
+object ParamCodec {
```

Comment:
`ParamCodec` is a typeclass for things that can be represented as a `Param`; ditto with `ParamSetCodec` and `ParamSet`. I created instances for the new target model but it will work with anything (although it's intended for data types written in Scala).